jOOQ-meta/src/main/java/org/jooq/util/AbstractDatabase.java

```diff
@@ -689,6 +691,21 @@ public void setSyntheticPrimaryKeys(String[] syntheticPrimaryKeys) {
     }
 
     @Override
+    public void setSyntheticIdentities(String[] syntheticIdentities) {
+        if (syntheticIdentities.length > 0 && !syntheticIdentities[0].isEmpty()) {
+            this.syntheticIdentities = syntheticIdentities;
+        }
+    }
+
+    @Override
+    public final boolean isSyntheticIdentity(ColumnDefinition columnDefinition) {
+        if (syntheticIdentities == null) {
+            return false;
+        }
+        return !filterExcludeInclude(singletonList(columnDefinition), null, syntheticIdentities, filters).isEmpty();
+    }
+
```

Comment:
Hmm, I prefer to have this logic similar to the existing getters / setters, and not perform any filtering. I will adapt this and move the filter logic out of the getter...